File: guild/src/test/java/com/os/operando/guild/QuintetTest.java
package com.os.operando.guild;

import org.junit.Test;

import java.time.LocalDate;
import java.util.Arrays;
import java.util.Calendar;
import java.util.List;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class QuintetTest {

    @Test
    public void test() {
        Calendar calendar = Calendar.getInstance();
        List<String> strings = Arrays.asList("quintet");
        Quintet<String, Integer, Boolean, Calendar, List<String>> quintet = Quintet.create("quintet", 0, false, calendar, strings);
        assertThat(quintet.getFirst(), is("quintet"));
        assertThat(quintet.getSecond(), is(0));
        assertThat(quintet.getThird(), is(false));
        assertThat(quintet.getFourth(), is(calendar));
        assertThat(quintet.getFive(), is(strings));
    }
}

File: guild-kotlin/src/test/kotlin/com/os/operando/guild/kt/QuintetTest.kt
package com.os.operando.guild.kt

import org.hamcrest.Matchers
import org.junit.Assert.*
import org.junit.Test
import java.time.LocalDateTime

class QuintetTest {

    @Test
    fun test() {
        val now = LocalDateTime.now()
        val quintet = 1 to 10L to true to now to "quintet"
        assertThat(quintet.first, Matchers.`is`(1))
        assertThat(quintet.second, Matchers.`is`(10L))
        assertThat(quintet.third, Matchers.`is`(true))
        assertThat(quintet.fourth, Matchers.`is`(now))
        assertThat(quintet.five, Matchers.`is`("quintet"))
    }

    @Test
    fun testToList() {
        val l = (0 to 1 to 2 to 3 to 4).toList()
        assertThat(l.size, Matchers.`is`(5))
        assertThat(l[0], Matchers.`is`(0))
        assertThat(l[1], Matchers.`is`(1))
        assertThat(l[2], Matchers.`is`(2))
        assertThat(l[3], Matchers.`is`(3))
        assertThat(l[4], Matchers.`is`(4))
    }
}